torchmetrics/metric.py

```diff
@@ -363,10 +439,15 @@ def _load_from_state_dict(
         error_msgs: List[str],
     ) -> None:
         """ Loads metric states from state_dict """
+
+        # only global rank 0 should be reloading the values present in the ``state_dict``
+        # as the state contains synced values across all progress_group
         for key in self._defaults:
             name = prefix + key
             if name in state_dict:
-                setattr(self, key, state_dict.pop(name))
+                value = state_dict.pop(name)
+                if os.getenv("GLOBAL_RANK", "0") == "0":
```

Comment:
Should we put this check inside a function? In case it needs to be overridden

Reply:
Done !